node/src/commands/deploy.js

```diff
@@ -2,41 +2,41 @@ const DeployUtils = require('../lib/deploy-utils');
 const logger = require('../lib/logger');
 const { options } = require('../config/constants');
 
-const { BLUEPRINT_ID } = options;
-
-const setConfigurationFromOptions = async (environmentVariables, environment, blueprintId) => {
-  const deployUtils = new DeployUtils();
-
-  if (environmentVariables && environmentVariables.length > 0) {
-    for (const config of environmentVariables) {
-      logger.info(
-        `Setting Environment Variable ${config.name} to be ${config.value} in environmentId: ${environment.id}`
-      );
-      await deployUtils.setConfiguration(environment, blueprintId, config.name, config.value, config.sensitive);
-    }
-  }
-};
+const { BLUEPRINT_ID, REVISION, ENVIRONMENT_NAME, PROJECT_ID } = options;
 
 const assertBlueprintExistsOnInitialDeployment = options => {
   if (!options[BLUEPRINT_ID]) throw new Error('Missing blueprint ID on initial deployment');
+  if (!options[REVISION]) throw new Error('Missing revision on initial deployment');
 };
 
+const getConfigurationChanges = environmentVariables =>
+  (environmentVariables || []).map(variable => ({
+    isSensitive: variable.sensitive,
+    name: variable.name,
+    value: variable.value,
+    scope: 'DEPLOYMENT',
+    type: 0 // supporting only environment variable type ATM
+  }));
+
 const deploy = async (options, environmentVariables) => {
   const deployUtils = new DeployUtils();
 
-  const { environmentName, projectId, organizationId, blueprintId } = options;
-
   logger.info('Waiting for deployment to start...');
-  let environment = await deployUtils.getEnvironment(environmentName, projectId);
+
+  const configurationChanges = getConfigurationChanges(environmentVariables);
+
+  let deployment;
+  let environment = await deployUtils.getEnvironment(options[ENVIRONMENT_NAME], options[PROJECT_ID]);
 
   if (!environment) {
+    logger.info('Initial deployment detected');
     assertBlueprintExistsOnInitialDeployment(options);
-    environment = await deployUtils.createEnvironment(environmentName, organizationId, projectId);
+    environment = await deployUtils.createAndDeployEnvironment(options, configurationChanges);
+    deployment = await deployUtils.getDeployment(environment.latestDeploymentLogId);
```

Comment:
this is not needed, we are only using the `id` anyway

Reply:
d7cec36